src/com/android/volley/toolbox/NetworkMonitor.java

```diff
@@ -57,7 +60,7 @@ public static void add(NetworkResponse response, ImageRequest request) {
             if (bandwidthKBPS < BANDWIDTH_HYSTERETIC_LOWER) {
                 lowBandwidth = true;
                 log("NETMON", "Low-Bandwidth flag On");
-            } else if (bandwidthKBPS > BANDWIDTH_HYSTERETIC_UPPER) {
+            } else if (bandwidthKBPS > BANDWIDTH_HYSTERETIC_UPPER && (!mIsRoaming || !mForceLowBandwidthIfRoaming)) {
```

Comment:
I think this is logically correct but a little confusing to read.

![image](https://cloud.githubusercontent.com/assets/1400053/5671607/2048e6c2-973e-11e4-9cf2-029d2b88afc2.png)

![image](https://cloud.githubusercontent.com/assets/1400053/5671615/38236970-973e-11e4-9e39-2a214dc23688.png)


Both `!mIsRoaming || (mIsRoaming && !mForceLowBandwidthIfRoaming)` and `!mIsRoaming || !mForceLowBandwidthIfRoaming` are the same and in the context of this statement will enable high bandwidth mode (as long as your connection is fast enough) unless you are both roaming *and* have lowBandwidthModeIfRoaming enabled.